manifests/init.pp

```diff
@@ -200,8 +200,8 @@
   String[1] $package_name                                          = $chrony::params::package_name,
   Optional[String] $package_source                                 = undef,
   Optional[String] $package_provider                               = undef,
-  $refclocks                                                       = [],
-  $peers                                                           = [],
+  Variant[Hash,Array[Variant[Hash,String[1]]]] $refclocks          = [],
+  Variant[Hash,Array[Stdlib::Host]] $peers                         = [],
   Variant[Hash,Array[Stdlib::Host]] $servers                       = {
```

Comment:
I wonder if these hashes can be further narrowed down with some type.

Reply:
I don't think so, I guess that would be something like

```
Variant[Hash[String,Variant[Array,Hash]],Array[Variant[Hash[String,Variant[Array,Hash]],String[1]]]] $refclocks
```

But that's so complicated that I don't think it would be a improvement. ;)